azurerm/resource_arm_subnet.go

```diff
@@ -301,3 +347,66 @@ func flattenSubnetIPConfigurations(ipConfigurations *[]network.IPConfiguration)
 
 	return ips
 }
+
+func expandSubnetDelegation(d *schema.ResourceData) []network.Delegation {
+	delegations := d.Get("delegation").([]interface{})
+	retDelegations := make([]network.Delegation, 0)
+
+	for _, deleValue := range delegations {
+		deleData := deleValue.(map[string]interface{})
+		deleName := deleData["name"].(string)
+		srvDelegations := deleData["service_delegation"].([]interface{})
+		srvDelegation := srvDelegations[0].(map[string]interface{})
+		srvName := srvDelegation["name"].(string)
+		srvActions := srvDelegation["actions"].([]interface{})
+
+		retSrvActions := make([]string, 0)
+		for _, srvAction := range srvActions {
+			srvActionData := srvAction.(string)
+			retSrvActions = append(retSrvActions, srvActionData)
+		}
+
+		retDelegation := network.Delegation{
+			Name: &deleName,
+			ServiceDelegationPropertiesFormat: &network.ServiceDelegationPropertiesFormat{
+				ServiceName: &srvName,
+				Actions:     &retSrvActions,
+			},
+		}
+
+		retDelegations = append(retDelegations, retDelegation)
+	}
+
+	return retDelegations
+}
+
+func flattenSubnetDelegation(delegations *[]network.Delegation) []interface{} {
+	if delegations == nil {
+		return []interface{}{}
+	}
+
+	retDeles := make([]interface{}, 0)
+
+	for _, dele := range *delegations {
+		retDele := make(map[string]interface{})
+		if dele.Name != nil {
+			retDele["name"] = *dele.Name
+		}
+
+		svcDeles := make([]interface{}, 0)
+		svcDele := make(map[string]interface{})
+		if dele.ServiceDelegationPropertiesFormat != nil && dele.ServiceName != nil {
```

Comment:
Could we be consistent with other resources and change this to:
```go
		if props := dele.ServiceDelegationPropertiesFormat; props != nil 
				if v := props.ServiceName; v != nil {
						svcDele["name"] = *v
				}
				if v := props.Actions; v != nil {
						svcDele["actions"] = *v
				}
		}
```

This is it is also a bit clearer why we are nil checking `ServiceDelegationPropertiesFormat`